pkg/netrc/reader.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019 The Go Authors. All rights reserved.
```

Comment:
Does this copyright allow copying code into your repo?  Also, why copy it instead of just using it as a module?  Did you have to make some tweaks to it?

Reply:
> Looks pretty good. I had a few small comments in various files.
> 
> As a first pass, this seems perfectly fine. I suspect you're going to want to refactor your utils package. It's a bit of a catch-all right now.
> 
> Also, you may want to go back and add tests for your internal files. There is quite a bit of code in there that could benefit from tests.
> 
> None of this is enough to prevent merging. I'd just like quick answers to the few questions I asked before I approve if you don't mind.

Yeah, the tests are coming. I wanted to try a few things first. I'll probably still end refactoring some of the way you enter the code, but for now it's ok.

Reply:
> Does this copyright allow copying code into your repo? Also, why copy it instead of just using it as a module? Did you have to make some tweaks to it?

I dont think there is a module you can use like that. It's built into the go binary. See here https://go.dev/src/cmd/go/internal/auth/netrc.go

Reply:
It might be good to cover your bases and just copy in the relevant lines from the governing [LICENSE](https://github.com/golang/go/blob/master/LICENSE) file for the original code.

Reply:
good idea. I added a license file and readme within that folder.